src/echo-app/src/preload/index.ts

```diff
@@ -1,39 +1,38 @@
-import { contextBridge } from 'electron'
-import { electronAPI } from '@electron-toolkit/preload'
 import path from 'path'
-import os from 'os'
-
 import { Module } from 'module'
 
-let sourceDir = path.resolve(os.homedir(), 'poestack-sage', 'plugins')
-if (import.meta.env.PRELOAD_VITE_PLUGIN_PATH) {
-  sourceDir = path.resolve('..', '..', import.meta.env.PRELOAD_VITE_PLUGIN_PATH)
-}
 
-const replaceResolvePath = (distPath: string) => {
-  if (distPath.endsWith('node_modules')) {
-    return distPath.replace(sourceDir, path.join(__dirname, '..', '..'))
-  }
-  return distPath.replace(sourceDir, path.join(__dirname, '..', '..', 'out', 'renderer'))
-}
-const orgResolvePath = Module['_resolveLookupPaths']
-Module['_resolveLookupPaths'] = function (request, parent) {
-  if (parent.filename?.startsWith(sourceDir)) {
-    // console.log('resolved lookup path', request, parent)
-    parent.filename = replaceResolvePath(parent.filename)
-    if (parent?.paths) {
-      parent.paths = parent.paths.map((p) => replaceResolvePath(p))
-    }
-    const res = orgResolvePath(request, parent)
-    // console.log('resolved lookup path', request, parent, res)
-    return res
-  } else {
-    const res = orgResolvePath(request, parent)
-    // console.log('resolved lookup path', request, parent, res)
-    return res
+const echoAppNodeModules: string = path.resolve(__dirname, '..', '..', 'node_modules')
+const echoCommonNodeModules: string = path.resolve(__dirname, '..', '..', '..', "echo-common", 'node_modules')
+console.log("echo-app-node-modules", echoAppNodeModules)
+console.log("echo-common-node-modules", echoCommonNodeModules)
+
+const orgResolvePath: unknown = Module['_resolveLookupPaths']
+Module['_resolveLookupPaths'] = function(request, parent) {
+
+  const orgResults: string[] = []
+  try {
+    orgResults.push(...orgResolvePath(request, parent))
+  } catch (error) {
   }
+
+  const mappedResults = orgResults.map((e) => e
+    .split("node_modules"))
+    .filter((e) => e.length == 2)
+    .flatMap((e) => [`${echoAppNodeModules}${e[1]}`, `${echoCommonNodeModules}${e[1]}`])
+
+  const results: string[] = []
+  results.push(...mappedResults)
+  results.push(...orgResults)
+  results.push(...[echoAppNodeModules, echoCommonNodeModules])
+
+
+  //console.log("req", request, parent, results)
+  return results
```

Comment:
It's going to load them from wherever https://github.com/PoeStack/poestack-sage/blob/main/src/echo-app/src/renderer/src/plugin-page.tsx#L67 this call loads them from. Module resolution doesn't dictate entry point

Reply:
Yeah true :D Thats all the magic behind this.
So this should work https://github.com/PoeStack/poestack-sage/blob/main/src/echo-app/src/renderer/src/plugin-page.tsx#L75

Reply:
Does this work, if for example `sage-common` has dependencies that `echo-app` or `echo-common` do not have?

Reply:
I didn't test it but it probably would work, the original resolution results are in the array so if it fails to find the lib in echo-app/echo-common it will try sage-common if sage-common was in the original array

Reply:
Okay I'm thinking about one problem in this architecture:
`sage-common` and `echo-common` have the same lib but `sage-common` has a different version as `echo-common`, then `sage-common` will not use the correct lib version.

Reply:
`echo-common` is dependent on `sage-common` and `echo-app` is dependent on `echo-common` if you have conflicting dep versions you will get warnings already

Reply:
`echo-common` is dependent on `sage-common`
Thats the point, if I understand your implementation `sage-common` resolve his `dependencies` from `echo-common` or `echo-app`. 
I'm trying to understand what 
```
 const mappedResults = orgResults.map((e) => e
    .split("node_modules"))
    .filter((e) => e.length == 2)
    .flatMap((e) => [`${echoAppNodeModules}${e[1]}`, `${echoCommonNodeModules}${e[1]}`])
```
does lol, but it seems, that echoAppNodeModules and echoCommonNodeModules have priority.

Reply:
Yeah but if NPM knows about the dep tree so if you try to install deps that have conflicting versions you will get warnings when you do the install before you ever even get to module resolution

Reply:
We have to resolve the dependency order correctly, this is why I prefer to only include the plugins. But we may extend this if we want to remove the `file:../../` stuff.

Reply:
The entire change is fixing the resolution order

Reply:
Yeah the goal is to remove as much `file:../../` as possible to fix Husky I didn't even see that it was in ggg-api and ts-ratchet I'll remove those as well

Reply:
Again :D You want `sage-common`, `ggg-api` and` ts-ratchet` to get their dependencies from `echo-common`? You can also see that this is currently implemented in `gradlew npmInstall`. It's shown like this there.

But this has nothing to do with the build order, because it is completely different. `sage-common`->`echo-common`->`echo-app` etc.

Reply:
> You want sage-common, ggg-api and ts-ratchet to get their dependencies from echo-common?
No, I would prefer not to have `ts-ratchet` pulling its deps from a project it is not dependent on and causing Husky to break.

Reply:
Okay, then we may have to test this case